internal/pkg/manifest/pipeline.go

```diff
@@ -110,7 +111,7 @@ func CreatePipeline(pipelineName string, provider Provider, stageNames []string)
 
 	stages := make([]PipelineStage, 0, len(stageNames))
 	for _, name := range stageNames {
-		stages = append(stages, PipelineStage{Name: name})
+		stages = append(stages, PipelineStage{Name: name, TestCommands: []string{}})
```

Comment:
Do we need to initialize the slice? I think we can keep it as `nil` assuming we want the field to be optional.

Reply:
Made this change and also added a test to pipeline_update_test to make sure if manifest.PipelineStage is `nil`, deploy.PipelineStage is also `nil`.